Sources/Engine/Base/Base.h

```diff
@@ -51,8 +53,29 @@ with this program; if not, write to the Free Software Foundation, Inc.,
     #endif
   #endif
 
-#endif  // defined _MSC_VER
+#elif (defined __linux__) 
+  #if (defined __ANDROID__) || (defined __android__) 
+    #error "Android current isn't supported"
+  #else
+    #define PLATFORM_LINUX
+  #endif
+#elif (defined __APPLE__)
+  #include "TargetConditionals.h"
+  #if TARGET_OS_MAC
```

Comment:
this doesn't work properly, as `Source/Engine/StdH.h` does `#undef TARGET_OS_MAC`